src/bin/pg_autoctl/cli_show.c

```diff
@@ -418,8 +423,43 @@ cli_show_state(int argc, char **argv)
 		exit(EXIT_CODE_QUIT);
 	}
 
+	/*
+	 * When dealing with a keeper node with a disabled monitor, we force the
+	 * --local option.
+	 */
+	if (!IS_EMPTY_STRING_BUFFER(config.pgSetup.pgdata) &&
+		ProbeConfigurationFileRole(config.pathnames.config) == PG_AUTOCTL_ROLE_KEEPER)
+	{
+		bool missingPgdataIsOk = true;
+		bool pgIsNotRunningIsOk = true;
+		bool monitorDisabledIsOk = true;
+
+		if (!keeper_config_read_file(&config,
+									 missingPgdataIsOk,
+									 pgIsNotRunningIsOk,
+									 monitorDisabledIsOk))
+		{
+			/* errors have already been logged */
+			exit(EXIT_CODE_BAD_CONFIG);
+		}
+
+		if (config.monitorDisabled)
+		{
+			log_info("Monitor is disabled, showing --local state");
+			(void) cli_show_local_state();
+			exit(EXIT_CODE_QUIT);
+		}
+	}
+
 	(void) cli_monitor_init_from_option_or_config(&monitor, &config);
 
+	if (optionGroupId != -1 && config.groupId != optionGroupId)
```

Comment:
Why are you disallowing querying other groups here?

Reply:
I guess this should instead be in `cli_show_local_state`.

Reply:
Good point, missed that entirely. Fixed now, thanks!